core/src/lib.rs

```diff
@@ -27,3 +27,17 @@ pub mod util;
 pub use primary_keys::MongoPrimaryKeys;
 mod foreign_keys;
 pub use foreign_keys::MongoForeignKeys;
+
+pub type WChar = u16;
+
+pub fn from_wchar_vec_lossy(v: Vec<u16>) -> String {
```

Comment:
I am not sure I want to add these utility methods in core. I am wondering if we could not have a crate to handle wstring, specialized in reading and writing String to Wchar buffer?

Also, yesterday I suggested to Oliver to add a struct and implementing into for the PowerBi tests.
[Something along this line](https://github.com/mongodb/mongo-odbc-driver/pull/72)
```
pub struct OutputBuffer<T> {
    pub output_buffer: Pointer,
    pub mut data_length: i32,
    pub buffer_length: i32,
}

impl Into<String> for OutputBuffer<WChar> {
    fn into(self) -> String {
        unsafe {
            String::from_utf16_lossy(slice::from_raw_parts(
                self.output_buffer as *const WChar,
                *self.data_length as usize,
            ))
        }
    }
}

impl Into<u16> for OutputBuffer<USmallInt> {
    fn into(self) -> u16 {
        unsafe { *(self.output_buffer as *mut u16) }
    }
}
```
Now, I'm thinking maybe we could do something similar in the driver itself.
Anyway, not for today, just a thought there.

Reply:
I like it. I do think we should use the widestring functions since we'll eventually want to support utf-32 anyway

Reply:
Yes, the internal impl is just a copy/paste of the function from the test. I did not look at making this part smart. I'm thinking I'll take the time to look into this a bit more during skunkwork maybe. I'd like to find a clean framework. The buffers handling is becoming a little too scattered. At least, it looks like it's not a completely dumb idea or I hope you'd say so :P

Reply:
Move to a crate